### DISCUSSION TITLE
For those who are not good at rust, integrate rustdesk into their own projects

### DISCUSSION THREAD

**Discussion by SmileZD** [rustdesk/rustdesk#642] [General]
A simple way

You can write a client in other languages, and then execute the CMD command to start the rustdesk-portable. You can change the password by adding the parameter --password startup in advance, and read the ID in the rustdesk configuration file to obtain the ID. the same is true for the control-client. You can add the parameter --connect to start the direct connection client. This client in other languages can be used at will, such as the account system.

I have built a c# client, which has only a few hundred KB. It always sends heartbeat packets to my server. When it receives the command to be remotely controlled, it checks or downloads the rustdesk-portable and sets the password in advance. After startup, it gets the ID and returns it to the server. The remote control terminal on the server receives the ID and the password set in advance. It can be remotely controlled normally. After the remote control is completed, it can also close the rustdesk portable version, Very convenient. The file is small and does not occupy memory during standby.

**Comment by SmileZD** [comment]
If you need to modify the server IP, you can directly copy and replace the configuration file that has changed the server IP to the target location before starting the software. If the ID is left blank, it will be automatically generated. The password can be set in advance, or you can use the --config parameter (not tried)

**Comment by Raki** [comment]
Hi @SmileZD ,
I tried running exe from **rustdesk-1.1.9-windows_x64-portable** with --password argument, but it is not running at all.
Can you tell me is this what you meant?

**Comment by congzhangzh** [comment]
But multi server does not work, if multi user connect to one server, then session manager is needed.

Maybe rustdesk can add a cmdline to switch the control rpc channel, the default hard code is /tmp/rustdesk?, which support sync password and so on, we need multi instance support :) @rustdesk

**Comment by MaxwellEdisons** [comment]
@SmileZD  “read the ID in the rustdesk configuration file to obtain the ID” Where is the location of the configuration file please？i don't have

**Reply by SmileZD** [reply]
C:\Users\Administrator\AppData\Roaming\RustDesk
Administrator=>Your windows count name

**Reply by MaxwellEdisons** [reply]
thank you!
